myriad-scheduler/src/main/java/com/ebay/myriad/Main.java

```diff
@@ -190,6 +196,27 @@ private void startNMInstances(Injector injector) {
       }
     }
 
+    /**
+     * Create NMProfile for any configured service
+     * @param injector
+     */
+    private void initJavaBasedTaskConfiguration(Injector injector) {
+      LOGGER.info("Initializing JavaBasedTaskConfiguration");
+      NMProfileManager profileManager = injector.getInstance(NMProfileManager.class);
+
+      Map<String, AuxTaskConfiguration> auxServicesConfigs = 
+          injector.getInstance(MyriadConfiguration.class).getAuxTaskConfigurations();
+      if (auxServicesConfigs != null) {
+        for (Map.Entry<String, AuxTaskConfiguration> entry : auxServicesConfigs.entrySet()) {
+          AuxTaskConfiguration config = entry.getValue();
+          final Double cpu = config.getCpus().or(AuxTaskConfiguration.DEFAULT_CPU);
+          final Double mem = config.getJvmMaxMemoryMB().or(AuxTaskConfiguration.DEFAULT_MEMORY);
+          
+          profileManager.add(new NMProfile(entry.getKey(), cpu, mem));
```

Comment:
I agree with @yufeldman if we're going to have multiple services (NM, JHS, etc) the proper way to do this is have a Generic Profile that can ideally a) determine the resources(cpu,mem,ports,disk)/attributes(hdfs,etc) needed, b) describe how to launch itself (addressing my other comment about code reuse), and maybe c) any service discovery primitives.  Also, I think most users would appreciate a less complicated configuration.